reark/src/main/java/io/reark/reark/data/store/ContentProviderStore.java

```diff
@@ -28,10 +31,45 @@
     @NonNull
     private final ContentObserver contentObserver = getContentObserver();
 
-    public ContentProviderStore(ContentResolver contentResolver) {
+    protected final PublishSubject<Pair<T, Uri>> updateSubject = PublishSubject.create();
+
+    public ContentProviderStore(@NonNull ContentResolver contentResolver) {
         this.contentResolver = contentResolver;
         this.contentResolver.registerContentObserver(
                 getContentUri(), true, contentObserver);
+
+        subscribe();
+    }
+
+    private void subscribe() {
+        updateSubject
+                .onBackpressureBuffer()
+                .observeOn(Schedulers.computation())
+                .subscribe(pair -> {
```

Comment:
This update function should be a utility - I can imagine someone wanting to use it for a different purposes and, for instance, make a static update subject to be shared across stores (or injected singleton). The it would also be easier to test.